Issue 57: Polish get started vignette (#59)
epinowcast/epidist: vignettes/epidist.Rmd

Unfortunately, estimating delays accurately from observational data is usually difficult.
In our experience, the two main issues are:

1. interval censoring, and
2. right truncation.

Don't worry if you've not come across these terms before.
First, in Section \@ref(data), we explain interval censoring and right truncation by simulating data like that we might observe during an ongoing infectious disease outbreak.
Then, in Section \@ref(fit), we show how `epidist` can be used to accurately estimate delay distributions by using a statistical model which properly accounts for these two issues.

If you would like more technical details, the `epidist` package implements models following best practices as described in @park2024estimating and @charniga2024best.

To run this vignette yourself, as well as the `epidist` package, you will need the `data.table`^[Note that to work with outputs from `epidist` you do not need to use `data.table`: any tool of your preference is suitable!], `purrr`, `ggplot2`, `gt`, and `dplyr` packages installed.

```{r load-requirements}
library(epidist)
library(data.table)
library(purrr)
library(ggplot2)

> library(gt)

Reviewer: We need this as a suggests in the description?

obs_cens <- obs |> observe_process()
```

> (ref:cens) Interval censoring of the primary and secondary event times obscures the delay times. Many exact delay times (which can take any value) have the same censored delay times (which are restricted to a number of days). While daily censoring is most common, `epidist` supports the primary and secondary events having other delay intervals.

Reviewer: Can you clarify this a bit to me? I'm not sure I follow the intent.

> (ref:cens) Interval censoring of the primary and secondary event times obscures the delay times. Many exact delay times (which can take any value) have the same censored delay times (which are restricted to a number of days). While daily censoring is most common, `epidist` supports the primary and secondary events having other delay intervals.

Reviewer: ```suggestion
(ref:cens) Interval censoring of the primary and secondary event times obscures the delay times. A common example of this is when events are reported as daily aggregates. While daily censoring is most common, `epidist` supports the primary and secondary events having other delay intervals.
```

I don't think this comment was addressed. I've reworded. If you feel strongly about the previous sentence please clarify why.

# Fit the model and compare estimates {#fit}

If we had access to the complete and unaltered `obs`, it would be simple to estimate the delay distribution.
However, with only access to `obs_cens_trunc_samp`, the delay distribution we observe is biased (Figure \@ref(fig:obs-est)) and we must use a statistical model.

> 

Author: Test

> (ref:obs-est) The histogram of delays from the complete retrospective data `obs_cens` match closely the underlying distribution, whereas those from `obs_cens_trunc_samp` are systematically biased. In this instance, while the complete, retrospective data are still subject to censoring, the extent of this issue is quite minor as compared with that from right truncation.

Reviewer: I don't think we should be downplaying censoring like this IMO. 

The teaching point is that the real-time data doesn't look like the underlying distribution and nor does the fully observed data but usually to a lesser degree. The key point being that both need a model to map to the underlying distribution. 

We can then add another teaching point which is that if we knew the infection time (and stress we likely never will) then observed censored daily data (i.e your daily histogram) we would have no bias (just increased obs error from the censoring and sampling variation).

Reviewer: One way to make this point (as I think @parksw3) is suggesting is to plot the mean and sd of the empirically observed data at different times vs the underlying true distribution. you could also add sampling variation to this if you wanted. This would then help you with some of your below points about "being close to the data"

Author: ![image](https://github.com/epinowcast/epidist/assets/29818116/3095e80f-2ed9-4210-af32-72a1db8f9317)

Here's the image that this is a caption for. It looks to me that the main issue here is the trunaction rather than the censoring (here we also have the sampling). So yes, it's a lesser degree, which the text does say. Perhaps I'd prefer to write "relatively" minor rather than "quite" minor.

Reviewer: My point here is we are still saying people should use a model in most cases

Author: > plot the mean and sd of the empirically observed data at different times vs the underlying true distribution. you could also add sampling variation to this if you wanted. 

I think this would add something to understanding. It might be quite a lot of code / complication for the getting started vignette, which I'd prefer to keep simple if possible. Happy to make another issue for this if you'd like.

Author: I am thinking of adding another sentence along the lines of "that said, we still think it's important to use a model to take into account censoring...". Is there a relevant little section or citation we could point to here for people who are interested? Could also just say in which situations it'd be more of an issue.

Author: @seabbs if there are any changes you'd like to make to get this over the line I'm happy for you go to into the branch.

Reviewer: > I think this would add something to understanding. It might be quite a lot of code / complication for the getting started vignette, which I'd prefer to keep simple if possible. Happy to make another issue for this if you'd like.

Yes that makes sense. I am suggesting this vs trying to pack in more context to the above plot.

> s there a relevant little section or citation we could point to here for people who are interested? Could also just say in which situations it'd be more of an issue.

The long delay paper and Kelly's paper both cover this. 

> "that said, we still think it's important to use a model to take into account censoring.

Yes agree. If we want to make this clearer we can just shift to using a slightly shorter delay as the censoring bias is absolute

Author: I've rewritten this caption. If there are specific changes you would like me to make please write them down and I will do them.

Author: Unsure if I should resolve here as the above caption is outdated.

Reviewer: Prior to resolving I think it would be useful to either respond to some of the above suggestions (i.e a shorter delay), move to issues, or adapt the current text in response to them

dplyr::bind_rows(
  obs |>
    dplyr::mutate(type = "Full data") |>
    dplyr::select(delay, type) |>

>     dplyr::mutate(delay = floor(delay)),

Author: Yes this made things clearer for me. That this is the floor of the delay, whereas the truncated delay is the difference between two floored numbers.
